add dolt_tag() procedure and dolt_tags system table docs (#784)
dolthub/docs: content/reference/sql/version-control/dolt-system-tables.md

```

## `dolt_tags`

> `dolt_tags` has all tags information created on the database.

Reviewer: ```suggestion
`dolt_tags` shows information for all active tags in the current database.
```

> DOLT_TAG() procedure can be used to INSERT and DELETE tags on the `dolt_remotes table.

Reviewer: A link here to the stored procedure docs would be awesome

Reviewer: Looks like there's a copy/paste error at the end of the sentence where it mentions the dolt_remotes table. There's also a missing backtick escape.

### Schema

```text
+----------+----------+
| Field    | Type     |
+----------+----------+
| tag_name | text     |
| tag_hash | text     |
| tagger   | text     |
| email    | text     |
| date     | datetime |
| message  | text     |
+----------+----------+
```

### Example Query

Create a tag using dolt_tag() stored procedure.

```sql
CALL DOLT_TAG('v2','head','-m','creating v2 tag');
```

```text
+--------+
| status |
+--------+
| 0      |
+--------+
```

Get all the tags.

```sql

> SELECT * FROM dolt_tags;

Reviewer: (minor) Does Andy's new `AS OF` support for system tables work with the tags table? If so, might be interesting to point out that customers can use AS OF to go back in time to see active tags (since any deleted tags don't show in the system table).

Author: I just checked, and `AS OF` is not supported for dolt_tags, I think I can add support for it.
